lib/module.js

```diff
@@ -144,52 +182,24 @@ Module._findPath = function(request, paths, isMain) {
     return Module._pathCache[cacheKey];
   }
 
-  var exts;
+  const exts = Object.keys(Module._extensions);
   const trailingSlash = request.length > 0 &&
                         request.charCodeAt(request.length - 1) === 47/*/*/;
+  const isAbsolute = !trailingSlash;
 
   // For each path
   for (var i = 0; i < paths.length; i++) {
     // Don't search further if path doesn't exist
     const curPath = paths[i];
     if (curPath && stat(curPath) < 1) continue;
-    var basePath = path.resolve(curPath, request);
-    var filename;
-
-    if (!trailingSlash) {
-      const rc = stat(basePath);
-      if (rc === 0) {  // File.
-        if (!isMain) {
-          filename = path.resolve(basePath);
-        } else {
-          filename = fs.realpathSync(basePath);
-        }
-      } else if (rc === 1) {  // Directory.
-        if (exts === undefined)
-          exts = Object.keys(Module._extensions);
-        filename = tryPackage(basePath, exts, isMain);
-      }
 
-      if (!filename) {
-        // try it with each of the extensions
-        if (exts === undefined)
-          exts = Object.keys(Module._extensions);
-        filename = tryExtensions(basePath, exts, isMain);
-      }
-    }
+    // If _basePath is a symlink, use the real path rather than the path of the
+    // symlink as cache key.
+    const _basePath = path.resolve(curPath, '/_' + request);
+    const basePath = path.resolve(curPath, request);
```

Comment:
The above two lines are contributing to v8 aborting optimization of `Module._findPath()` because of the `const` usage. They should be reverted to `var` for now.